### PR TITLE
Remove the `@git_cmd` macro

### PR DESCRIPTION
**Pull request by DilumAluthge**
See also: #30 

This is not part of the public API, so it is not a breaking change.

### CODE REVIEW

**Comment by DilumAluthge** [comment]
cc: @fredrikekre @gunnarfarneback

**Review by giordano — APPROVED** [review]
Looks ok, besides the version number (I can't really tell whether to bump patch or minor...)

File: test/runtests.jl [hunk 1]
```diff
@@ -34,19 +34,6 @@ end
         @test isdir("Git.jl")
         @test isfile(joinpath("Git.jl", "Project.toml"))
     end
```
Comment (GunnarFarneback) [inline test/runtests.jl:36]: Out of curiosity, is there a reason for doing the same block of tests twice?

Comment (DilumAluthge) [reply test/runtests.jl:36]: IIRC, we want to call the `git()' function multiple times as part of our "make sure the package is thread-safe" tests? @giordano

Comment (GunnarFarneback) [reply test/runtests.jl:36]: That looks suspiciously sequential for testing thread-safety.

Comment (giordano) [reply test/runtests.jl:36]: No, I suspect that's an artifact of refactoring the tests.

Comment (DilumAluthge) [reply test/runtests.jl:36]: We should add a test that spawns multiple tasks and makes sure that the clones all work normally. And then at the end, checks that still no environment variables have leaked.

**Pull request merged**